docs/sql-ref-number-pattern.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: global
+title: Number patterns
+displayTitle: Number Patterns for Formatting and Parsing
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+Spark uses pattern letters in the following table for number parsing and formatting:
+
+|Symbol|Meaning|Examples|
+|------|-------|--------|
+|**9**|digit position (can be dropped if insignificant)|9999|
```

Comment:
```suggestion
|**9**|Position for a digit; leading/trailing zeros are trimmed.|9999|
```

Reply:
I noticed that snowflake does not simply trim the leading/trailing zeros, but replace with spaces. Is it a common behavior in other databases?